source/common/upstream/upstream_impl.cc

```diff
@@ -94,6 +94,15 @@ parseClusterSocketOptions(const envoy::api::v2::Cluster& config,
         cluster_options,
         Network::SocketOptionFactory::buildTcpKeepaliveOptions(parseTcpKeepaliveConfig(config)));
   }
+  // Cluster socket_options trump cluster manager wide.
+  if (bind_config.socket_options().size() + config.upstream_bind_config().socket_options().size() >
+      0) {
+    auto socket_options = config.upstream_bind_config().socket_options().size() > 0
```

Comment:
Do we want to override or union here? I think override as you're doing, because that gives you negative logic, i.e. the ability to eliminate the more general config socket options with the more specific.

Reply:
Agreed. I think the facility will be used infrequently enough and/or with few enough options that it won't be burdensome to duplicate options from the CM config to per-cluster config and being able to mask out seems like an important feature which we'd lose.